apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -77,58 +61,46 @@ export class GeneratePreviewUsecase {
         );
       }
 
-      let previewDataResult = {
+      let previewTemplateData = {
         variablesExample: {},
         controlValues: {},
       };
 
       for (const [controlKey, controlValue] of Object.entries(sanitizedValidatedControls)) {
-        // previewControlValue is the control value that will be used to render the preview
-        const previewControlValue = controlValue;
-        // variableControlValue is the control value that will be used to extract the variables example
-        let variableControlValue = controlValue;
-        if (isStringTipTapNode(variableControlValue)) {
-          try {
-            variableControlValue = transformMailyContentToLiquid(JSON.parse(variableControlValue));
-          } catch (error) {
-            console.log(error);
-          }
-        }
+        const variables = buildVariables(variableSchema, controlValue, this.logger);
+        const processedControlValues = this.fixControlValueInvalidVariables(controlValue, variables.invalidVariables);
 
-        const variables = this.processControlValueVariables(variableControlValue, variableSchema);
-        const processedControlValues = this.fixControlValueInvalidVariables(previewControlValue, variables.invalid);
-
-        const validVariableNames = variables.valid.map((variable) => variable.name);
+        const validVariableNames = variables.validVariables.map((variable) => variable.name);
         const variablesExampleResult = pathsToObject(validVariableNames, {
           valuePrefix: '{{',
           valueSuffix: '}}',
         });
 
-        previewDataResult = {
-          variablesExample: _.merge(previewDataResult.variablesExample, variablesExampleResult),
+        previewTemplateData = {
+          variablesExample: _.merge(previewTemplateData.variablesExample, variablesExampleResult),
           controlValues: {
-            ...previewDataResult.controlValues,
+            ...previewTemplateData.controlValues,
             [controlKey]: isObjectTipTapNode(processedControlValues)
               ? JSON.stringify(processedControlValues)
               : processedControlValues,
           },
         };
       }
 
-      const finalVariablesExample = this.buildVariable(workflow, previewDataResult, commandVariablesExample);
+      const mergedVariablesExample = this.mergeVariablesExample(workflow, previewTemplateData, commandVariablesExample);
       const executeOutput = await this.executePreviewUsecase(
         command,
         stepData,
-        finalVariablesExample,
-        previewDataResult.controlValues
+        mergedVariablesExample,
+        previewTemplateData.controlValues
       );
 
       return {
         result: {
           preview: executeOutput.outputs as any,
           type: stepData.type as unknown as ChannelTypeEnum,
         },
-        previewPayloadExample: finalVariablesExample,
+        previewPayloadExample: mergedVariablesExample,
```

Comment:
As per lexicon, these are the [preview control data](https://www.notion.so/novuhq/Product-Terminology-Lexicon-d6ef923953744bc1b967ba8f297dee7f?pvs=4#15ad03ad838f80cbb958e9906eddee46). Let's add a TODO for the renaming in a follow up PR.

I also have a task in my list, to remove the nesting added by the `result` attribute from this endpoint.

Reply:
agree :)